factory/alchemy.py

```diff
@@ -39,15 +43,69 @@ class SQLAlchemyModelFactory(base.Factory):
     class Meta:
         abstract = True
 
+    @classmethod
+    def _generate(cls, strategy, params):
+        # Original params are used in _get_or_create if it cannot build an
+        # object initially due to an IntegrityError being raised
+        cls._original_params = params
+        return super(SQLAlchemyModelFactory, cls)._generate(strategy, params)
+
+    @classmethod
+    def _get_or_create(cls, model_class, *args, **kwargs):
+        session = cls._meta.sqlalchemy_session
+
+        key_fields = {}
+        for field in cls._meta.sqlalchemy_get_or_create:
+            if field not in kwargs:
+                raise errors.FactoryError(
+                    "sqlalchemy_get_or_create - "
+                    "Unable to find initialization value for '%s' in factory %s" %
+                    (field, cls.__name__))
+            key_fields[field] = kwargs.pop(field)
+
+        obj = session.query(model_class).filter_by(
+            *args, **key_fields).one_or_none()
```

Comment:
Can a test be added to verify this behavior?

Reply:
Sure! Let me know what you think.

Reply:
I like it, it’s short and effective, thanks!